### ISSUE TITLE
VoiceOver read characters when we set `obscureText: true` in `TextField`

### ISSUE THREAD

**Issue by QuncCccccc** [flutter/flutter#190320]
### Steps to reproduce

1. turn on VoiceOver
2. focus on a TextField with obscureText: true
3. type characters

### Expected results

VoiceOver does not read the plaintext password aloud

### Actual results

VoiceOver reads the plaintext password aloud. I do see the virtual keyboard pops up but somehow the recording doesn't show up.

### Code sample

<details open><summary>Code sample</summary>

```dart
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Semantics(headingLevel: 1, child: Text('$pageTitle Demo'))),
      body: ListView(
        children: const <Widget>[
          TextField(
            key: Key('enabled password'),
            decoration: InputDecoration(labelText: 'Password'),
            obscureText: true,
          ),
          TextField(
            key: Key('disabled password'),
            decoration: InputDecoration(labelText: 'Password'),
            enabled: false,
            obscureText: true,
          ),
        ],
      ),
    );
  }
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/2919da34-0f2c-41b4-b4a3-5a0cb05a4dcd

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by chunhtai** [comment]
TYUI is the soft keyboard key locations that the a11y box is focused on, are you sure it is announcing the content of the textfield?

**Comment by QuncCccccc** [comment]
What is TYUI? I filed this bug in last iOS a11y bug bash, when I updated to iOS 27 and tried the sample code, the screen reader just announce whatever I typed:)

**Comment by chunhtai** [comment]
It looks like me that the the voice over in the video is announcing "T" "Y" "U" "I" as a result of focusing the soft keyboard key.

**Comment by QuncCccccc** [comment]
Ah I see. Thanks for the explanation! That's a good point! I just tried it again and seems typing will be announced as well. I also tried it on iOS 26.6 and see the same issue.

This is the complete sample code I used:
<details><summary>Code sample</summary>

```dart
import 'package:material_ui/material_ui.dart';

void main() {
  runApp(const Issue190320App());
}

class Issue190320App extends StatelessWidget {
  const Issue190320App({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Semantics(
            headingLevel: 1,
            child: const Text('Issue 190320 Demo'),
          ),
        ),
        body: ListView(
          padding: const EdgeInsets.all(16.0),
          children: const <Widget>[
            TextField(
              key: Key('enabled password'),
              decoration: InputDecoration(labelText: 'Password'),
              obscureText: true,
            ),
            SizedBox(height: 16),
            TextField(
              key: Key('disabled password'),
              decoration: InputDecoration(labelText: 'Password'),
              enabled: false,
              obscureText: true,
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

https://github.com/user-attachments/assets/2c87e4fb-1005-401c-a039-109b5b5589c4

**Comment by chunhtai** [comment]
It still feel to me it is annocing the key in softkeyboard, not react to the text that get's typed into the textfield. Can you double check with any native password field whether they do the same?
